Tracking/include/Tracking/Reco/GreedyAmbiguitySolver.h

```diff
@@ -0,0 +1,210 @@
+//--- Refit with backward propagation ---//
+//#include "Acts/TrackFitting/KalmanFitter.hpp"
+
+//-- Ambiguity Solving --//
+//#include "Tracking/Reco/AmbiguitySolver.h"
+//#include "Tracking/Reco/ScoreBasedAmbiguitySolver.h"
+
+// GSF
+//#include "Acts/TrackFitting/GaussianSumFitter.hpp"
+//#include "Acts/Propagator/MultiEigenStepperLoop.hpp"
+
+//--- Tracking ---//
+#include "Tracking/Event/Measurement.h"
+#include "Tracking/Event/Track.h"
+#include "Tracking/Reco/TrackExtrapolatorTool.h"
+#include "Tracking/Sim/IndexSourceLink.h"
+#include "Tracking/Sim/MeasurementCalibrator.h"
+#include "Tracking/Sim/TrackingUtils.h"
+
+//--- Interpolated magnetic field ---//
+#include "Tracking/Sim/BFieldXYZUtils.h"
+
+namespace tracking {
+namespace reco {
+
+/**
+ * Minimal example of a processor.
+ *
+ * This processor will loop over all of the ECal hits in an event and
+ * print out their details.
+ */
+class GreedyAmbiguitySolver final : public TrackingGeometryUser {
+ public:
+  /**
+   * Constructor.
+   *
+   * @param name Name for this instance of the class.
+   * @param process The Process class associated with EventProcessor,
+   * provided by the framework.
+   */
+  GreedyAmbiguitySolver(const std::string& name, framework::Process& process);
+
+  /// Destructor
+  ~GreedyAmbiguitySolver();
+
+  void onProcessStart() override;
+
+  /**
+   * onNewRun is the first function called for each processor
+   * *after* the conditions are fully configured and accessible.
+   * This is where you could create single-processors, multi-event
+   * calculation objects.
+   */
+  void onNewRun(const ldmx::RunHeader& rh) override;
+
+  /**
+   *
+   */
+  void onProcessEnd() override;
+
+  /**
+   * Configure the processor using the given user specified parameters.
+   *
+   * The user specified parameters that are availabed are defined
+   * in the python configuration class. Look at the my_processor.py
+   * module of the EventProc python for the python structure.
+   *
+   * @param parameters Set of parameters used to configure this processor.
+   */
+  void configure(framework::config::Parameters& parameters) override;
+
+  /**
+   * Process the event and put new data products into it.
+   *
+   * @param event The event to process.
+   */
+  void produce(framework::Event& event) override;
+
+ private:
+  /// Maximum amount of shared hits per track.
+  std::uint32_t maximumSharedHits_{1};
+  /// Maximum number of iterations
+  std::uint32_t maximumIterations_{1000};
+
+  /// Minimum number of measurement to form a track.
+  std::size_t nMeasurementsMin_{7};
+
+  std::string out_trk_collection_{"TaggerTracksClean"};
+
+  std::string trackCollection_{"TaggerTracks"};
+
+  std::string measCollection_{"DigiTaggerSimHits"};
+
+  struct State {
+    std::size_t numberOfTracks{};
+
+    std::vector<int> trackTips;
+    std::vector<float> trackChi2;
+    std::vector<std::vector<std::size_t>> measurementsPerTrack;
+
+    // TODO consider boost 1.81 unordered_flat_map
```

Comment:
just FYI I have a PR open that will make an OS update and that will get to newer boost https://github.com/LDMX-Software/docker/pull/103